### PR TITLE
llvm: also write *-apple-macosx* config files

### LINKED ISSUE #197278: llvm: Support all `-macosx` target triples
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```console
% brew gist-logs llvm@19
Error: No logs.

% brew config
HOMEBREW_VERSION: 4.4.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 824efa8836dc226aa92dbbf7404c1b4a66707cca
Last commit: 7 days ago
Core tap JSON: 10 Nov 22:51 UTC
Core cask tap JSON: 10 Nov 22:51 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit dunno
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: N/A
Rosetta 2: false

% brew doctor
Your system is ready to brew.
```

Note that the system is a clean virtual machine with a newly installed homebrew.

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

https://github.com/Homebrew/homebrew-core/pull/196094 introduced new logic for finding the SDK root.

There are a few problems with it, however, including:
1. It conflates SDK root and deployment target (the version in the target triples is the deployment target, and doesn't have much to do with SDK versions).
2. It does not support "`macosx`" target triples (such as `arm64-apple-macosx`).

Discovered originally in https://github.com/rust-lang/cc-rs/issues/1278.

### What happened (include all command output)?

Using Clang from the `llvm` package with the `-target arm64-apple-macosx10.7` flag fails to link (or compile), because the logic for using a different SDK is not correct.

```
ld: library 'System' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

### What did you expect to happen?

Using any macOS target triple supported by Clang compiles, as long as there's at least one installed macOS SDK.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install llvm@19
echo "int main() { return 0; }" > foo.c
/opt/homebrew/opt/llvm@19/bin/clang -target arm64-apple-macosx10.7 foo.c
```

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Fixes #197278.

Also, remove the CLT requirement for pouring bottles. Many users use
this for the libraries, which don't need the CLT. We can add it back if
users report getting tripped up by this.

### CODE REVIEW

File: Formula/l/llvm.rb [hunk 1]
```diff
@@ -504,19 +496,26 @@ def write_config_files(macos_version, kernel_version, arch)
     arches = Set.new([:arm64, :x86_64])
     arches << arch
 
+    systems = %w[darwin macosx]
```
Comment (Bo98) [inline Formula/l/llvm.rb:499]:
```suggestion
    systems = {
      darwin: kernel_version,
      macosx: macos_version,
    }
```

File: Formula/l/llvm.rb [hunk 2]
```diff
@@ -504,19 +496,26 @@ def write_config_files(macos_version, kernel_version, arch)
...
     arches.each do |target_arch|
-      target_triple = "#{target_arch}-apple-darwin#{kernel_version}"
-      (clang_config_file_dir/"#{target_triple}.cfg").atomic_write <<~CONFIG
-        --sysroot=#{MacOS::CLT::PKG_PATH}/SDKs/MacOSX#{macos_version}.sdk
-      CONFIG
+      systems.each do |system|
+        config_file = "#{target_arch}-apple-#{system}#{kernel_version}.cfg"
```
Comment (Bo98) [inline Formula/l/llvm.rb:502]:
```suggestion
      systems.each do |system, version|
        config_file = "#{target_arch}-apple-#{system}#{version}.cfg"
```

**Review by Bo98 — COMMENTED** [review]
You'll want `macosx#{macos_version}` and `darwin#{kernel_version}`

**Comment by carlocab** [comment]
> You'll want `macosx#{macos_version}` and `darwin#{kernel_version}`

Will do, thanks. Also, just remembered:

What about `*-apple-macos*`? I think that's a valid triple too. 🙃

**Comment by Bo98** [comment]
My upstream patch calls `Triple.getOSTypeName` which should already normalise that to `macosx`.

File: Formula/l/llvm.rb [hunk 3]
```diff
@@ -504,19 +496,29 @@ def write_config_files(macos_version, kernel_version, arch)
...
   end
 
   def post_install
     return unless OS.mac?
-    return if (clang_config_file_dir/"#{Hardware::CPU.arch}-apple-darwin#{OS.kernel_version.major}.cfg").exist?
 
-    write_config_files(MacOS.version.major, OS.kernel_version.major, Hardware::CPU.arch)
+    config_files = %w[darwin macosx].map do |system|
+      clang_config_file_dir/"#{Hardware::CPU.arch}-apple-#{system}#{OS.kernel_version.major}.cfg"
+    end
```
Comment (Bo98) [inline Formula/l/llvm.rb:518]: This needs updating too - forgot about that

File: Formula/l/llvm.rb [hunk 4]
```diff
@@ -449,16 +446,11 @@ def install
...
-        11 => 20,
-        12 => 21,
-        13 => 22,
-        14 => 23,
-        15 => 24,
-      }.each do |macos_version, kernel_version|
-        write_config_files(macos_version, kernel_version, Hardware::CPU.arch)
+      MacOSVersion::SYMBOLS.each_key do |v|
+        macos_version = MacOSVersion.new(v)
+        next if macos_version < "10.14" # system headers are in `/usr` before this.
```
Comment (Bo98) [inline Formula/l/llvm.rb:451]: Thinking about this more, < 10.14 will just fallback to unversioned triple so we should probably move this conditional into `write_config_files` and write `--sysroot=/` `if macos_version < "10.14"`

Comment (carlocab) [reply Formula/l/llvm.rb]: Thanks!

**Review by Bo98 — APPROVED** [review]
Thanks!

**Pull request merged**